### ISSUE TITLE
Not counting the vast majority of supporting reads in one direction

### ISSUE THREAD

**Issue by macko-tipor** [AstraZeneca-NGS/VarDict#73]
Hi,

I am using VarDict to call somatic variants in paired tumor-normal samples.
The libraries were prepared using QIASeq Targeted DNA Panels kit and paired-end sequenced on a NextSeq instrument.
The BCL files were converted into uBAMs. Adapters were marked by Picard Tools, then mapped BAMs were generated via a pipe from uBAMs by BWA-MEM and Picard Tools.
Consensus reads were created by fgbio using UMIs, adapters were marked again, the consensus reads were filtered by fgbio, and then mapped by applying the same pipe as before. Primers were clipped by bamclipper, BAMs were validated by Picard Tools and all SAM validation errors were fixed.
Variants were called from BAMs contaning no error.

Moreover, a samtools flagstat file shows the following info about the tumor sample:

7698164 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 secondary
0 + 0 supplementary
0 + 0 duplicates
7677316 + 0 mapped (99.73% : N/A)
7698164 + 0 paired in sequencing
3849082 + 0 read1
3849082 + 0 read2
7677316 + 0 properly paired (99.73% : N/A)
7656494 + 0 with itself and mate mapped
20822 + 0 singletons (0.27% : N/A)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)

The following command was used for variant calling:

"$VARDICT" \\
    -th 48 \\
    -G "$Refseq" \\
    -f 0.10 \\
    -r 8 \\
    -3 \\
    -b ""$INPUT_TUMOR"|"$INPUT_NORMAL"" \\
    -N "$SAMPLE_TUMOR" \\
    -c 1 \\
    -S 2 \\
    -E 3 \\
    "$TARGET_INTERVALS_VARDICT" | "$DIR_VARDICT"/testsomatic.R | "$DIR_VARDICT"/var2vcf_paired.pl \\
        -f 0.10 \\
        -r 8 \\
        -N ""$SAMPLE_TUMOR"|"$SAMPLE_NORMAL"" \\
        > "$OUTPUT"

The step has completed without error. However, the depth data of the variants were extremely different from those seen in a browser.

VarDict data:

STATUS=LikelySomatic;SAMPLE=172T_S8;TYPE=SNV;SHIFT3=0;MSI=1.000;MSILEN=1;SSF=0;SOR=139.85833;LSEQ=TGTGGATTGTGCAAAATTAA;RSEQ=ACGACTCCTGAAGGGTAAGT	GT:DP:VD:ALD:RD:AD:AF:BIAS:PMEAN:PSTD:QUAL:QSTD:SBF:ODDRATIO:MQ:SN:HIAF:ADJAF:NM	0/1:130:60:1,59:2,67:69,60:0.4615:1,1:40.8:1:44:1:1:1.75379:60:59:0.4758:0:2.4	0/0:167:1:0,1:1,165:166,1:0.006:1,0:40:0:45:0:1:0:60:2:0.0061:0:7

Browser data in normal sample:

Type                Base	Count	% of Total	Mean Quality
(match)		A	        318	        98.1		        43.0
(mismatch)	C	            2	          0.6		        45.0
(mismatch)	N	            4	          1.2		          2.0
Total		                324	      100.0	                42.5

Browser data in tumor sample:

Type		Base	Count	% of Total	Mean Quality
(match)		A	        126	        50.6		        42.5
(mismatch)	C	        121	        48.6		        44.2
(mismatch)	N	            1	          0.4		          2.0
(mismatch)	T	            1	          0.4		        13.0
Total		                249	      100.0                 43.0

The most surprising thing is the difference in the number of supporting reads with different directions.
For example, I counted in the browser the number of forward and reverse reads supporting either the alternative or the reference variant in the tumor sample in this position.
Out of the 121 mismatches, 56 are in forward and 65 are in reverse reads. According to VarDict, the respective numbers are 1 and 59.
Out of the 126 matches, 57 are in forward and 69 are in reverse reads. VarDict ALD shows 2 and 67.

Similar phenomena can be seen in other positions, too.

In summary, it seems to me that the vast majority of the forward reads are not counted by VarDict in my samples. What can be the reason for this behavior? Perhaps filtering or uninformative reads?

Thank you for your help in advance!

**Comment by mjafin** [comment]
Hi @macko-tipor, thanks for the detailed report. Some time ago there was an effort to address double counting of fragments where the two reads of a fragment overlap and both support the variant. 

In this case the fragment should only be counted once but the logistics of dealing with overlapping reads got a bit messy. In an effort to try and save memory we implemented a simple scheme where only one of the reads would be analysed (say, forward strand) for the overlapping part, ignoring the other. 

This kind of works but can lead to false strand bias like in your case. The (lower) depth reported by Vardict is probably more indicative though as it's the estimated true fragment depth.

In the latest release there are now three options:
- ignore the problem (where variant depth will look similar to browser data but doesn't reflect fragments)
- count only forward read evidence (`-u`)
- count only R1 read evidence (`-UN`)

Have you looked into these?

**Comment by macko-tipor** [comment]
Hi,

Thank you for your answer.

I wanted to use the default setting (neither -u nor -UN, that is unique mode disabled) which I expected to result in depth values similar to browser data. But it was this default setting that led to counting reads only in one direction. My understanding is that results like mine should be obtained by using any of the two mentioned options.

Of course, I agree that one piece of evidence should not be counted twice. 
I am going to test the two options now. I will also test re-running the command with default setting after clipping the overlapping part of the pairs. I suppose that I should get similar decreased depth values in all three cases.

**Comment by mjafin** [comment]
Hi @macko-tipor, just check that the version of VarDict you're running indeed has the three options as what's now behind `-u` used to be the default option for a while.

**Comment by macko-tipor** [comment]
I am working with VarDict-1.5.1 now.
Is -u for getting depth data by counting both overlapping reads for this version?

Anyway, I will check and download the latest version.

**Comment by mjafin** [comment]
You will want `1.5.2` or later. Just grab the latest (`1.5.4`)

**Comment by macko-tipor** [comment]
Yes, version 1.5.1 does not recognize -u.

It seems to me, that there is no more problem here. 
I just used an old version while reading the latest manual on the net.

Tomorrow I will download version 1.5.4.

Than you very much for your help!

**Comment by mjafin** [comment]
Great, let us know if your problem isn't resolved by the new version and the available read counting options.
